polling_stations/apps/core/slack_client.py

```diff
@@ -0,0 +1,59 @@
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
+import os
+from typing import Optional, Dict, Any
+
+
+class SlackChannelNotFoundError(Exception):
+    pass
+
+
+class SlackClient:
+    def __init__(self, token: Optional[str] = None, channel: str = "bots"):
+        self.token = token or os.getenv("SLACK_TOKEN")
+        if not self.token:
+            raise ValueError(
+                "Slack token not provided and SLACK_TOKEN not found in environment"
+            )
+
+        self.client = WebClient(token=self.token)
+        self.channel = channel
+        self.channel_id = self.get_channel_id_from_name(self.channel)
+
+    def get_channel_id_from_name(self, channel_name: str) -> Optional[str]:
+        conversation_id = None
+        try:
+            conversations_response = self.client.conversations_list()
+            for channel in conversations_response.data["channels"]:
+                if channel["name"] == channel_name:
+                    conversation_id = channel["id"]
+                    return conversation_id
+        except SlackApiError as e:
+            print(f"Error: {e}")
```

Comment:
Tiny nit-pick here, but might be a useful pattern to think about. It's tricky to test this output as it's going to `stdout`. It might be nicer to set `self.stdout = sys.stdout` in the `__init__` and allow passing a stdout into the class. Then in tests we can pass a `StringIO` and read from it. All calls to `print` can be converted to `self.stdout.write`.